test/end-to-end/getRedirectionURL.test.js

```diff
@@ -377,5 +375,152 @@ describe("getRedirectionURL Tests", function () {
                 assert.equal(newUserCheck, "thirdpartypasswordless-true");
             });
         });
+
+        describe("No Redirection", function () {
+            describe("Email Password Recipe", function () {
+                let browser;
+                let page;
+
+                before(async function () {
+                    await backendBeforeEach();
+
+                    await fetch(`${TEST_SERVER_BASE_URL}/startst`, {
+                        method: "POST",
+                    }).catch(console.error);
+
+                    browser = await puppeteer.launch({
+                        args: ["--no-sandbox", "--disable-setuid-sandbox"],
+                        headless: true,
+                    });
+
+                    page = await browser.newPage();
+                    // We need to set the localStorage value before the page loads to ensure ST initialises with the correct value
+                    await page.evaluateOnNewDocument(() => {
+                        localStorage.setItem("disableRedirectionAfterSuccessfulSignInUp", "true");
+                        localStorage.removeItem("isNewUserCheck");
+                    });
+
+                    await clearBrowserCookiesWithoutAffectingConsole(page, []);
+                });
+
+                after(async function () {
+                    await browser.close();
+                    await fetch(`${TEST_SERVER_BASE_URL}/after`, {
+                        method: "POST",
+                    }).catch(console.error);
+                    await fetch(`${TEST_SERVER_BASE_URL}/stopst`, {
+                        method: "POST",
+                    }).catch(console.error);
+
+                    await screenshotOnFailure(this, browser);
+                });
+
+                it("should not do any redirection after successful sign up", async function () {
+                    await Promise.all([
+                        page.goto(`${TEST_CLIENT_BASE_URL}/auth?authRecipe=emailpassword`),
+                        page.waitForNavigation({ waitUntil: "networkidle0" }),
+                    ]);
+
+                    await toggleSignInSignUp(page);
+                    const urlBeforeSignUp = await page.url();
+                    await defaultSignUp(page);
+                    const urlAfterSignUp = await page.url();
+
+                    const newUserCheck = await page.evaluate(() => localStorage.getItem("isNewUserCheck"));
+                    assert.equal(newUserCheck, "emailpassword-true");
+                    assert.equal(urlBeforeSignUp, urlAfterSignUp);
+                });
+            });
+
```

Comment:
what happens if you sign in -> get redirected to the home page -> run localStorage.setItem("disableRedirectionAfterSuccessfulSignInUp", "true") -> go back to the auth page. I assume this will behave weirdly in some way? Cause the auth page by default detects if a session exists, and if it does, it tries to redirect, but we are returning null, and then on each render of the auth page, it will try to do this.

Reply:
If a user attempts to access the /auth page while the `getRedirectionURL` function returns `null`, the result will be a blank page without further action.

You are correct that every time the Auth page loads, it will try to redirect. This is not a problem because this only happens per render and returning `null` doesn't cause any more re-renders.

Reply:
I wonder if we should instead show the login UI as usual in this case as opposed to showing a blank page? If the user does not return `null` from `getRedirectionURL`, then it makes sense to show a blank page other the page might flicker. But if they are returning `null`, it's very weird to just show a blank page.